src/main/groovy/org/transmartproject/batch/concept/ConceptsTablesWriter.groovy

```diff
@@ -0,0 +1,174 @@
+package org.transmartproject.batch.concept
+
+import groovy.util.logging.Slf4j
+import groovy.xml.MarkupBuilder
+import org.springframework.batch.item.ItemWriter
+import org.springframework.batch.repeat.RepeatStatus
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.beans.factory.annotation.Value
+import org.springframework.jdbc.core.simple.SimpleJdbcInsert
+import org.springframework.stereotype.Component
+import org.transmartproject.batch.beans.JobScopeInterfaced
+import org.transmartproject.batch.clinical.db.objects.Tables
+import org.transmartproject.batch.db.DatabaseUtil
+import org.transmartproject.batch.facts.ClinicalFactsRowSet
+import org.transmartproject.batch.support.SecureObjectToken
+
+/**
+ * Database writer of patient rows, based on FactRowSets
+ */
+@Component
+@JobScopeInterfaced
+@Slf4j
+class ConceptsTablesWriter implements ItemWriter<ClinicalFactsRowSet> {
+
+    @Autowired
+    ConceptTree conceptTree
+
+    @Autowired
+    SecureObjectToken secureObjectToken
+
+    @Value(Tables.CONCEPT_DIMENSION)
+    private SimpleJdbcInsert dimensionInsert
+
+    @Value(Tables.I2B2)
+    private SimpleJdbcInsert i2b2Insert
+
+    @Value(Tables.I2B2_SECURE)
+    private SimpleJdbcInsert i2b2SecureInsert
+
+    @SuppressWarnings('PrivateFieldCouldBeFinal')
+    @Lazy
+    private String metadataXml = { generateMetadataXml() }()
+
+    @Value("#{jobParameters['STUDY_ID']}")
+    String studyId
+
+    @Override
+    void write(List<? extends ClinicalFactsRowSet> items) throws Exception {
+        //TODO Get new concepts from items and move it to own itemProcessor
```

Comment:
Rather than put a cryptic TODO (which, btw, will probably never be addressed), it would be better to explain that is inserting the new patients put in the ConceptTree during the step processor rather than looking at the items. It took me a while to figure that out.